apollo-router/src/plugins/headers.rs

```diff
@@ -347,8 +349,9 @@ where
                     req.supergraph_request
                         .headers()
                         .iter()
-                        .filter(|(name, _)| matching.is_match(name.as_str()))
```

Comment:
good question, I think reordering would be pretty much the same, but let's godbolt it!

https://godbolt.org/z/9KodG6hco it looks like both closures are kept around for some reason 🤔

Reply:
That's an interesting tool.

I guess it defaults to debug though, because it looks like it decides that filter_1 and filter_2 are the same thing in release:
https://godbolt.org/z/qEsGsd7b3

Reply:
Yes, you have to pass --release explicitely

Reply:
oh gosh i forgot to release the parking brake 😅